src/pybind/mgr/dashboard/frontend/src/app/shared/directives/trim.directive.ts

```diff
@@ -0,0 +1,19 @@
+import { Directive, HostListener } from '@angular/core';
+import { NgControl } from '@angular/forms';
+
+@Directive({
+  selector: '[cdTrim]'
+})
+export class TrimDirective {
+  constructor(private ngControl: NgControl) {}
+
+  @HostListener('input', ['$event.target.value'])
+  onInput(value) {
+    this.setValue(value);
+  }
+
+  setValue(value: string): void {
+    value = typeof value === 'string' ? value.trim() : value;
```

Comment:
Looks more handy, but is doing the same. Personally i always use lodash instead of plain JS.
```suggestion
    value = _.isString(value) ? value.trim() : value;
```

Reply:
I'm now using `lodash` to check is `value` is a `string`.